benchexec/tools/sikraken.py

```diff
@@ -0,0 +1,53 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+"""
+This class serves as BenchExec tool adaptor for Sikraken for Test-Comp
+See baseclass documentation at https://github.com/sosy-lab/benchexec/blob/main/benchexec/tools/template.py
+"""
+import os
+from benchexec.tools.sv_benchmarks_util import get_data_model_from_task, ILP32, LP64
+import benchexec.tools.template
+import benchexec.result as result
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    BenchExec tool-info for Sikraken
+    """
+
+    def name(self):
+        return "Sikraken" 
+    
+    def project_url(self):
+        return "https://github.com/echancrure/Sikraken" 
+        
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("sikraken.sh", subdir="bin")
+
+    def version(self, executable):
+        return self._version_from_tool(executable, "-v")  
+    
+    """
+    #Assuming that the default working directory is the root of the project (i.e. sikraken/) we don't need this
+    def working_directory(self, executable):
+        executableDir = os.path.dirname(executable)
+        return executableDir
+    """
```

Comment:
If the tool requires the working directory to be the project's root directory, it would be good to implement this method appropriately, such that users do not have to fulfill this requirement manually.

(But in general, it is better practice for tools to be executable independently of the current working directory. If you change your tool to support this, please remove this commented-out code.)

Reply:
agreed and removed